test/ping_test.go

```diff
@@ -41,6 +41,7 @@ func TestPingSentToTLSConnection(t *testing.T) {
 	tc.CaFile = opts.TLSCaCert
 
 	opts.TLSConfig, _ = server.GenTLSConfig(&tc)
+	opts.TLSTimeout = float64(5 * time.Second)
```

Comment:
I believe that TLSTimeout is a float representing the number of seconds. So 0.5 means 0.5s or 500ms. To set it to 5sec, you should set it to 5.0. 5*time.Second is a huge number which would translate to: `1388888h53m20s`